Sources/MapboxMaps/Annotations/Generated/CircleAnnotationManager.swift

```diff
@@ -206,36 +206,23 @@ public class CircleAnnotationManager: AnnotationManager {
         }
     }
 
-    // MARK: - Selection Handling -
+    // MARK: - Tap Handling -
 
     /// Set this delegate in order to be called back if a tap occurs on an annotation being managed by this manager.
     public weak var delegate: AnnotationInteractionDelegate? {
         didSet {
-            if delegate != nil {
-                setupTapRecognizer()
-            } else {
-                guard let view = view, let recognizer = tapGestureRecognizer else { return }
-                view.removeGestureRecognizer(recognizer)
-                tapGestureRecognizer = nil
+            if delegate != nil && oldValue == nil {
+                singleTapGestureRecognizer?.addTarget(self, action: #selector(handleTap(_:)))
+            } else if delegate == nil && oldValue != nil {
+                singleTapGestureRecognizer?.removeTarget(self, action: #selector(handleTap(_:)))
```

Comment:
It'd be simpler and less error-prone to just set up the target/action once in init and check for a non-nil delegate in the handler.

Reply:
If you do this, you won't even need to store a ref to the gesture recognizer